### PR TITLE
Add support for new WDAC API

### PR DESCRIPTION
**Pull request by PaulHigin**
# PR Summary

Implement new WDAC policy script file block option.

## PR Context

WDAC API now supports the ability to allow/block PowerShell script files, whereas before the option was only to allow full/constrained language modes.  This PR implements the new API with fallback to original system lock down behavior.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/security/wldpNativeMethods.cs [hunk 1]
```diff
@@ -12,6 +12,32 @@
 
 namespace System.Management.Automation.Security
 {
+    /// <summary>
+    /// System wide policy enforcement for a specific script file.
+    /// </summary>
+    public enum SystemScriptFileEnforcement
```
[the remark this answers is not in the document]
Comment (PaulHigin) [reply src/System.Management.Automation/security/wldpNativeMethods.cs:18]: It is not duplicated.  The API type is translated into a PowerShell type that is more meaningful.

**Comment by PaulHigin** [comment]
Ping - @anamnavi, @TravisEz13, @daxian-dbw

**Comment by daxian-dbw** [comment]
Sorry for the delay. Will do the review next Monday 😄

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

File: src/System.Management.Automation/resources/SecuritySupportStrings.resx [hunk 2]
```diff
@@ -153,4 +153,10 @@
   <data name="InvalidSessionKey" xml:space="preserve">
     <value>Invalid session key data.</value>
   </data>
+  <data name="ScriptFileBlockedBySystemPolicy" xml:space="preserve">
+    <value>Script file, '{0}', is blocked from running by system policy.</value>
+  </data>
+  <data name="UnknownSystemScriptFileEnforcement" xml:space="preserve">
+    <value>An unknown script file policy enforcement value was returned: {0}</value>
```
Comment (daxian-dbw) [inline src/System.Management.Automation/resources/SecuritySupportStrings.resx:160]:
```suggestion
    <value>An unknown script file policy enforcement value was returned: {0}.</value>
```

**Review by daxian-dbw — APPROVED** [review]
LGTM, with a minor comment.

**Comment by TravisEz13** [comment]
There was a breaking change in Windows 11 22H2 related to this feature which defaults CI script policy to block.  We should not backport this to 7.2 until we verify that this issue has been resolved.

**Pull request merged**